File: bindings/android/src/main/java/com/nativeview/NativeViewConfig.java
package com.nativeview;

/**
 * Window creation options for Android; passed to native {@code nv_window_cfg_t} via JNI.
 *
 * <p>Field names and types match the desktop {@code NvWindowCfg} style for binding parity.
 */
public class NativeViewConfig {

    /** Window title; may be {@code null} (native default applies). */
    public String title;

    public int width = 1024;
    public int height = 768;

    /** When {@code true}, initial layout may allow resize where the host supports it. */
    public boolean resizable = true;

    /** When {@code true}, enables WebView developer tooling where supported. */
    public boolean devtools;
}

File: bindings/ios/Sources/NativeViewIOS/App/NativeViewConfig.swift
// SPDX-License-Identifier: Apache-2.0

import Foundation

/// Window shell options for ``NativeViewApp/createWindow(config:)`` (parity with desktop bindings).
public struct NativeViewConfig: Sendable {
    public var title: String
    public var width: Int
    public var height: Int
    public var resizable: Bool
    public var devtools: Bool

    public init(
        title: String = "",
        width: Int = 390,
        height: Int = 844,
        resizable: Bool = true,
        devtools: Bool = false
    ) {
        self.title = title
        self.width = width
        self.height = height
        self.resizable = resizable
        self.devtools = devtools
    }
}
